mllib/src/main/scala/org/apache/spark/ml/feature/Bucketizer.scala

```diff
@@ -201,9 +184,13 @@ final class Bucketizer @Since("1.4.0") (@Since("1.4.0") override val uid: String
 
   @Since("1.4.0")
   override def transformSchema(schema: StructType): StructType = {
-    if (isBucketizeMultipleColumns()) {
+    ParamValidators.checkExclusiveParams(this, "inputCol", "inputCols")
```

Comment:
The problem with trying to use a general method like this is that it's hard to capture model-specific requirements.  This currently misses checking to make sure that exactly one (not just <= 1) of each pair is available, plus that all of the single-column OR all of the multi-column Params are available.  (The same issue occurs in https://github.com/apache/spark/pull/20146 )  It will also be hard to check these items and account for defaults.

I'd argue that it's not worth trying to use generic checking functions here.

Reply:
my initial implementation (with @hhbyyh's comments) was more generic and checked what you said. After, @MLnick and @viirya asked to switch to a more generic approach which is the current you see. I'm fine with either of those, but I think we need to choose one way and go in that direction, otherwise we just loose time.

Reply:
I see.  I'll see if I can come up with something which is generic but handles these other checks.